nodescraper/cli/helper.py

```diff
@@ -331,3 +332,77 @@ def generate_reference_config(
     plugin_config.plugins = plugins
 
     return plugin_config
+
+
+def generate_reference_config_from_logs(
+    path: str, plugin_reg: PluginRegistry, logger: logging.Logger
+) -> PluginConfig:
+    """Parse previous log files and generate plugin config with populated analyzer args
+
+    Args:
+        path (str): path to log files
+        plugin_reg (PluginRegistry): plugin registry instance
+        logger (logging.Logger): logger instance
+
+    Returns:
+        PluginConfig: instance of plugin config
+    """
+    found = find_datamodel_and_result(path)
+    plugin_config = PluginConfig()
+    plugins = {}
+    for dm, res in found:
+        result_path = Path(res)
+        res_payload = json.loads(result_path.read_text(encoding="utf-8"))
+        task_res = TaskResult(**res_payload)
+        dm_path = Path(dm)
+        dm_payload = json.loads(dm_path.read_text(encoding="utf-8"))
+        plugin = plugin_reg.plugins.get(task_res.parent)
+        data_model = plugin.DATA_MODEL.model_validate(dm_payload)
+
+        if not plugin.ANALYZER_ARGS:
+            logger.warning(
+                "Plugin: %s does not support reference config creation. No analyzer args defined.",
+                task_res.parent,
+            )
+            continue
+
+        args = None
+
+        try:
+            args = plugin.ANALYZER_ARGS.build_from_model(data_model)
+        except NotImplementedError as nperr:
+            logger.info(nperr)
+            continue
+        plugins[task_res.parent] = {"analysis_args": {}}
+        plugins[task_res.parent]["analysis_args"] = args.model_dump(exclude_none=True)
```

Comment:
This could be consolidated to a single line
```suggestion
        plugins[task_res.parent] = {"analysis_args": args.model_dump(exclude_none=True)}
```